File: LoginSystem.java
import java.io.*;
import java.util.*;
import org.json.simple.*;
import org.json.simple.parser.*;

public class LoginSystem {
    private static final String USERS_FILE = "users.json";
    private User currentUser;
    private JSONParser parser = new JSONParser();

    public static void main(String[] args) {
        LoginSystem loginSystem = new LoginSystem();
        loginSystem.run();
    }

    public void run() {
        Scanner scanner = new Scanner(System.in);
        initializeUsersFile();

        while (true) {
            System.out.println("\n=== Login System ===");
            System.out.println("1. Register");
            System.out.println("2. Login");
            System.out.println("3. Exit");
            System.out.print("Choose an option: ");

            int choice = scanner.nextInt();
            scanner.nextLine(); // consume newline

            switch (choice) {
                case 1:
                    registerUser(scanner);
                    break;
                case 2:
                    loginUser(scanner);
                    break;
                case 3:
                    System.out.println("Exiting...");
                    return;
                default:
                    System.out.println("Invalid option!");
            }

            if (currentUser != null) {
                showUserMenu(scanner);
            }
        }
    }

    private void initializeUsersFile() {
        File file = new File(USERS_FILE);
        if (!file.exists()) {
            try (FileWriter writer = new FileWriter(USERS_FILE)) {
                writer.write("{}");
            } catch (IOException e) {
                System.err.println("Error creating users file: " + e.getMessage());
            }
        }
    }

    @SuppressWarnings("unchecked")
    private void registerUser(Scanner scanner) {
        System.out.println("\n--- Registration ---");
        System.out.print("Enter username: ");
        String username = scanner.nextLine().trim();

        JSONObject users = loadUsers();
        if (users.containsKey(username)) {
            System.out.println("Username already exists!");
            return;
        }

        System.out.print("Enter password: ");
        String password = scanner.nextLine();

        System.out.println("\nSelect account type:");
        System.out.println("1. Tenant (Privilege 1)");
        System.out.println("2. Property Owner (Privilege 2)");
        System.out.println("3. Admin (Privilege 3)");
        System.out.print("Enter choice: ");
        int accountType = scanner.nextInt();
        scanner.nextLine(); // consume newline

        if (accountType < 1 || accountType > 3) {
            System.out.println("Invalid account type!");
            return;
        }

        JSONObject newUser = new JSONObject();
        newUser.put("password", password);
        newUser.put("privilege", accountType);

        users.put(username, newUser);
        saveUsers(users);

        System.out.println("Registration successful!");
    }

    private void loginUser(Scanner scanner) {
        System.out.println("\n--- Login ---");
        System.out.print("Enter username: ");
        String username = scanner.nextLine().trim();

        System.out.print("Enter password: ");
        String password = scanner.nextLine();

        JSONObject users = loadUsers();
        if (!users.containsKey(username)) {
            System.out.println("User not found!");
            return;
        }

        JSONObject user = (JSONObject) users.get(username);
        String storedPassword = (String) user.get("password");
        long privilege = (long) user.get("privilege");

        if (storedPassword.equals(password)) {
            currentUser = new User(username, password, (int) privilege);
            System.out.println("Login successful!");
            System.out.println("Welcome, " + username + "!");
            System.out.println("Account type: " + getAccountTypeName((int) privilege));
        } else {
            System.out.println("Incorrect password!");
        }
    }

    private void showUserMenu(Scanner scanner) {
        while (currentUser != null) {
            System.out.println("\n--- User Menu ---");
            System.out.println("1. View Profile");
            System.out.println("2. Logout");
            
            // Admin-specific options
            if (currentUser.getPrivilege() == 3) {
                System.out.println("3. List All Users");
            }
            
            System.out.print("Choose an option: ");
            int choice = scanner.nextInt();
            scanner.nextLine(); // consume newline

            switch (choice) {
                case 1:
                    viewProfile();
                    break;
                case 2:
                    currentUser = null;
                    System.out.println("Logged out successfully!");
                    return;
                case 3:
                    if (currentUser.getPrivilege() == 3) {
                        listAllUsers();
                    } else {
                        System.out.println("Invalid option!");
                    }
                    break;
                default:
                    System.out.println("Invalid option!");
            }
        }
    }

    private void viewProfile() {
        System.out.println("\n--- Your Profile ---");
        System.out.println("Username: " + currentUser.getUsername());
        System.out.println("Account Type: " + getAccountTypeName(currentUser.getPrivilege()));
        System.out.println("Privilege Level: " + currentUser.getPrivilege());
    }

    @SuppressWarnings("unchecked")
    private void listAllUsers() {
        JSONObject users = loadUsers();
        System.out.println("\n--- All Users ---");
        
        for (Object key : users.keySet()) {
            String username = (String) key;
            JSONObject user = (JSONObject) users.get(username);
            long privilege = (long) user.get("privilege");
            
            System.out.println("Username: " + username + 
                               " | Type: " + getAccountTypeName((int) privilege) +
                               " | Privilege: " + privilege);
        }
    }

    private String getAccountTypeName(int privilege) {
        switch (privilege) {
            case 1: return "Tenant";
            case 2: return "Property Owner";
            case 3: return "Admin";
            default: return "Unknown";
        }
    }

    private JSONObject loadUsers() {
        try (FileReader reader = new FileReader(USERS_FILE)) {
            return (JSONObject) parser.parse(reader);
        } catch (IOException | ParseException e) {
            System.err.println("Error loading users: " + e.getMessage());
            return new JSONObject();
        }
    }

    private void saveUsers(JSONObject users) {
        try (FileWriter writer = new FileWriter(USERS_FILE)) {
            writer.write(users.toJSONString());
        } catch (IOException e) {
            System.err.println("Error saving users: " + e.getMessage());
        }
    }

    private static class User {
        private String username;
        private String password;
        private int privilege;

        public User(String username, String password, int privilege) {
            this.username = username;
            this.password = password;
            this.privilege = privilege;
        }

        public String getUsername() {
            return username;
        }

        public String getPassword() {
            return password;
        }

        public int getPrivilege() {
            return privilege;
        }
    }
}

File: Delete Later/LoginSystem.py
import json
import os

USERS_FILE = 'users.json'

class User:
    def __init__(self, username, password, privilege):
        self.username = username
        self.password = password
        self.privilege = privilege

    def get_username(self):
        return self.username

    def get_password(self):
        return self.password

    def get_privilege(self):
        return self.privilege


class LoginSystem:
    def __init__(self):
        self.current_user = None
        self.initialize_users_file()

    def run(self):
        while True:
            print("\n=== Login System ===")
            print("1. Register")
            print("2. Login")
            print("3. Exit")
            choice = input("Choose an option: ")

            if choice == '1':
                self.register_user()
            elif choice == '2':
                self.login_user()
            elif choice == '3':
                print("Exiting...")
                break
            else:
                print("Invalid option!")

            if self.current_user:
                self.show_user_menu()

    def initialize_users_file(self):
        if not os.path.exists(USERS_FILE):
            with open(USERS_FILE, 'w') as file:
                json.dump({}, file)

    def load_users(self):
        try:
            with open(USERS_FILE, 'r') as file:
                return json.load(file)
        except Exception as e:
            print(f"Error loading users: {e}")
            return {}

    def save_users(self, users):
        try:
            with open(USERS_FILE, 'w') as file:
                json.dump(users, file, indent=4)
        except Exception as e:
            print(f"Error saving users: {e}")

    def register_user(self):
        print("\n--- Registration ---")
        username = input("Enter username: ").strip()

        users = self.load_users()
        if username in users:
            print("Username already exists!")
            return

        password = input("Enter password: ")
        print("\nSelect account type:")
        print("1. Tenant (Privilege 1)")
        print("2. Property Owner (Privilege 2)")
        print("3. Admin (Privilege 3)")
        account_type = input("Enter choice: ")

        if account_type not in {'1', '2', '3'}:
            print("Invalid account type!")
            return

        users[username] = {
            "password": password,
            "privilege": int(account_type)
        }
        self.save_users(users)
        print("Registration successful!")

    def login_user(self):
        print("\n--- Login ---")
        username = input("Enter username: ").strip()
        password = input("Enter password: ")

        users = self.load_users()
        if username not in users:
            print("User not found!")
            return

        user_data = users[username]
        if user_data["password"] == password:
            self.current_user = User(username, password, user_data["privilege"])
            print("Login successful!")
            print(f"Welcome, {username}!")
            print(f"Account type: {self.get_account_type_name(user_data['privilege'])}")
        else:
            print("Incorrect password!")

    def show_user_menu(self):
        while self.current_user:
            print("\n--- User Menu ---")
            print("1. View Profile")
            print("2. Logout")
            if self.current_user.get_privilege() == 3:
                print("3. List All Users")
            choice = input("Choose an option: ")

            if choice == '1':
                self.view_profile()
            elif choice == '2':
                self.current_user = None
                print("Logged out successfully!")
            elif choice == '3' and self.current_user.get_privilege() == 3:
                self.list_all_users()
            else:
                print("Invalid option!")

    def view_profile(self):
        print("\n--- Your Profile ---")
        print(f"Username: {self.current_user.get_username()}")
        print(f"Account Type: {self.get_account_type_name(self.current_user.get_privilege())}")
        print(f"Privilege Level: {self.current_user.get_privilege()}")

    def list_all_users(self):
        users = self.load_users()
        print("\n--- All Users ---")
        for username, data in users.items():
            privilege = data["privilege"]
            print(f"Username: {username} | Type: {self.get_account_type_name(privilege)} | Privilege: {privilege}")

    def get_account_type_name(self, privilege):
        return {
            1: "Tenant",
            2: "Property Owner",
            3: "Admin"
        }.get(privilege, "Unknown")


if __name__ == "__main__":
    login_system = LoginSystem()
    login_system.run()
